hypothesis-python/src/hypothesis/searchstrategy/types.py

```diff
@@ -27,16 +27,18 @@
 
 import hypothesis.strategies as st
 from hypothesis.errors import InvalidArgument, ResolutionFailed
-from hypothesis.internal.compat import text_type, integer_types
+from hypothesis.internal.compat import PY2, text_type
 
 
 def type_sorting_key(t):
     """Minimise to None, then non-container types, then container types."""
     if not isinstance(t, type):
         raise InvalidArgument('thing=%s must be a type' % (t,))
     if t is None or t is type(None):  # noqa: E721
-        return -1
-    return issubclass(t, collections.abc.Container)
+        return (-1, repr(t))
+    if PY2:
```

Comment:
I'm pretty 👎 on `if PY2` checks outside of `compat.py`. Couldn't this be a `from hypothesis.internal.compat import Container`?